fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/Repository.java

```diff
@@ -133,6 +143,75 @@ public BinaryObject createBinary(URI path, String slug, InputStream content, Str
 		BinaryObject binary = new BinaryObject(newPid, this, repositoryObjectDataLoader);
 		return binary;
 	}
+	
+	/**
+	 * Creates an event for the specified object.
+	 * 
+	 * @param eventPid
+	 *            the PID of the event to add
+	 * @param model
+	 *            Model containing properties of this event. Must only contain
+	 *            the properties for one event.
+	 * @return URI of the event created
+	 * @throws FedoraException
+	 */
+	public PremisEventObject createPremisEvent(PID eventPid, Model model) throws FedoraException {
+
+		try (FcrepoResponse response = getClient().put(eventPid.getRepositoryUri())
+				.body(RDFModelUtil.streamModel(model), "text/turtle")
+				.perform()) {
+
+			return new PremisEventObject(PIDs.get(response.getLocation()), this, repositoryObjectDataLoader);
+		} catch (IOException e) {
+			throw new FedoraException("Unable to create premis event for " + eventPid, e);
+		} catch (FcrepoOperationFailedException e) {
+			throw ClientFaultResolver.resolve(e);
```

Comment:
ClientFaultResolver.resolve inspects the exception that comes back from the fcrepoClient request and rethrows it as a FedoraException (our local exception for stuff that goes wrong in fedora) or preferably, one of its subclasses.  At the moment this just involves looking at the status code of the response, but in the past it has been a bit more involved since you had to inspect fedora 3's response bodies to figure out what went wrong.

This lets us keep the FcrepoClient library from percolating up into the rest of the repository and allows us to react to exceptions that indicate that, for example, an authentication error occurred instead of a generic exception.